src/utils/HTTPTransport.ts

```diff
@@ -0,0 +1,79 @@
+enum Methods {
+  GET = 'GET',
+  POST = 'POST',
+  PUT = 'PUT',
+  DELETE = 'DELETE',
+}
+
+type TRequestData = Record<string, string | number>;
+
+export type TRequestOptions = {
+  method?: Methods
+  headers?: Record<string, string>
+  timeout?: number
+  data?: unknown
+};
+
+function queryStringify(data: TRequestData) {
+  if (typeof data !== 'object') {
+    throw new Error('Data must be object');
+  }
+
+  const keys = Object.keys(data);
+  return keys.reduce((result, key, index) => `${result}${key}=${data[key]}${index < keys.length - 1 ? '&' : ''}`, '?');
+}
+
+export class HTTPTransport {
+  public get = (url: string, options: TRequestOptions = {}): Promise<XMLHttpRequest> => {
+    if (!!options.data) {
+      url = `${url}${queryStringify(options.data as TRequestData)}`;
+    }
+    return this.request(url, { ...options, method: Methods.GET });
+  };
+
+  public post = (url: string, options = {}): Promise<XMLHttpRequest> => this.request(url, { ...options, method: Methods.POST });
+
+  public put = (url: string, options = {}): Promise<XMLHttpRequest> => this.request(url, { ...options, method: Methods.PUT });
+
+  public delete = (url: string, options = {}): Promise<XMLHttpRequest> => this.request(url, { ...options, method: Methods.DELETE });
+
+  request = (url: string, options: TRequestOptions = {}): any => {
```

Comment:
Тут возвращается не any, а Promise, поэтому стоит это явно указать
```suggestion
  request = (url: string, options: TRequestOptions = {}): Promise<unknown> => {
```